persist/fs/files.go

```diff
@@ -42,11 +42,103 @@ var timeZero time.Time
 
 const (
 	dataDirName       = "data"
+	snapshotDirName   = "snapshots"
 	commitLogsDirName = "commitlogs"
 )
 
 type fileOpener func(filePath string) (*os.File, error)
 
+// FilesetFile represents a set of Fileset files for a given block start
+type FilesetFile struct {
+	BlockStart time.Time
+	Files      []string
+}
+
+// FilesetFilesSlice is a slice of FilesetFile
+type FilesetFilesSlice []FilesetFile
```

Comment:
suggestion(take/leave): I tend to prefer not naming types based on the backing implementation unless I have no better idea for the name. That way if you decide to change the implementation, the refactor is more localized. i.e. `FilesetFiles` conveys as much information as `FilesetFilesSlice` and needs fewer changes if/when you change the implementation to something like `map[UnixNano]FilesetFile`. 

Completely driven by preference obv, so feel free to ignore if you disagree.

Reply:
Agreed, I think I added slice to the end because we already have functions in this package correspondingly named FilesetFiles and SnapshotFiles and I liked how the existing names read when they're used outside the package.

I considered using the word "list" but it sounds like a lie cause its not really a linked list

Reply:
Fair enough.